examples/rust/tcp-ping-pong.rs

```diff
@@ -334,13 +350,18 @@ impl TcpClient {
 
             // Push data.
             {
-                let sga: demi_sgarray_t = mksga(&mut self.libos, BUFFER_SIZE, fill_char)?;
-                if let Err(e) = push_and_wait(&mut self.libos, self.sockqd, &sga) {
-                    freesga(&mut self.libos, sga);
+                self.sga = Some(mksga(&mut self.libos, BUFFER_SIZE, fill_char)?);
+                if let Err(e) = push_and_wait(
+                    &mut self.libos,
+                    self.sockqd,
+                    &self.sga.expect("should be a valid sgarray"),
+                ) {
+                    freesga(&mut self.libos, self.sga.expect("should be a valid sgarray"));
```

Comment:
This can be removed and no expect is required then. The drop() function can clean it up like the sockets.

Reply:
Done.